File: lumina/core/tests/smoke_test.rs
// TEST-BOOTSTRAP:STUB
//! Smoke test — confirms the chosen test stack (cargo test + rstest +
//! pretty_assertions + proptest) is wired correctly. Passes on first run.

#[test]
fn arithmetic_works() {
    assert_eq!(1 + 1, 2);
}

#[tokio::test]
async fn async_runtime_works() {
    let val: u32 = async { 42 }.await;
    assert_eq!(val, 42);
}

File: lumina/web/src/__tests__/smoke.test.ts
// TEST-BOOTSTRAP:STUB
// Smoke test — confirms `bun test` is wired correctly for the Vue SPA target.
// Should pass on first run. Invoke: `bun test` (from lumina/web/).

import { describe, expect, test } from 'bun:test'

describe('smoke', () => {
  test('arithmetic works', () => {
    expect(1 + 1).toBe(2)
  })

  test('async resolution works', async () => {
    const val = await Promise.resolve(42)
    expect(val).toBe(42)
  })
})
